docs/element-definitions.md

```diff
@@ -1353,6 +1356,339 @@ the `inferred` link tells the user that the element was created based on some
+      "attributes": {
+        "href": "/users"
+      },
+      "content": [
+        {
+          "element": "transition",
+          "content": [
+            {
+              "element": "httpTransaction",
+              "attributes": {
+                "authSchemes": [
+                  {
+                    "element": "Custom Basic Auth"
+                  }
+                ]
+              }
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Token Authentication Scheme (Base API Element)
+
+This describes an authentication scheme that uses a token as a way to authenticate and authorize. The token MAY exist as an HTTP header field or a query parameter.
+
+- One of
+    - `httpHeaderName` (string)
+    - `queryParameterName` (string)
+
+When used as a query parameter, an HREF Variable is not required to be defined within the scope of the resource or transition, but is rather infered from the used token authentications scheme.
+
+#### Properties
+
+- `element`: Token Authentication Scheme (string, fixed)
+- `content` (array[Member Element])
+
+#### Example
+
+This example shows a custom token authentication scheme being defined as `Custom Token Auth` that uses a query parameter for the token. This scheme is then used on an HTTP transaction within a resource. Please note this example is incomplete for the sake of keeping it short.
+
+```json
+{
+  "element": "category",
+  "meta": {
+    "classes": ["api"]
+  },
+  "content": [
+    {
+      "element": "category",
+      "meta": {
+        "classes": ["authSchemes"]
+      },
+      "content": [
+        {
+          "element": "Token Authentication Scheme",
+          "meta": {
+            "id": "Custom Token Auth"
+          },
+          "content": [
+            {
+              "element": "member",
+              "content": {
+                "key": {
+                  "element": "string",
+                  "content": "queryParameterName"
+                },
+                "value": {
+                  "element": "string",
+                  "content": "api_key"
+                }
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "element": "resource",
+      "attributes": {
+        "href": "/users"
+      },
+      "content": [
+        {
+          "element": "transition",
+          "content": [
+            {
+              "element": "httpTransaction",
+              "attributes": {
+                "authSchemes": [
+                  {
+                    "element": "Custom Token Auth"
+                  }
+                ]
+              }
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Oauth2 Scheme (Base API Element)
+
+This describes an authentication scheme that uses Oauth2 as defined in [RFC 6749](https://tools.ietf.org/html/rfc6749).
+
+The element MAY have the following members to define additional information about the Oauth2 implementation.
+
+- `scopes` (array[string])
```

Comment:
This is the difficulty in describing refract data structures :) It is member element where the key is scope and the content is an array of string elements, but we are actually wanting the member elements.

This is the same situation for username and password for basic auth.